BedrockCore.cpp

```diff
@@ -64,6 +64,98 @@ bool BedrockCore::isTimedOut(unique_ptr<BedrockCommand>& command) {
     return false;
 }
 
+BedrockCore::RESULT BedrockCore::prePeekCommand(unique_ptr<BedrockCommand>& command) {
+    AutoTimer timer(command, BedrockCommand::PREPEEK);
+    BedrockServer::ScopedStateSnapshot snapshot(_server);
+    command->lastPeekedOrProcessedInState = _server.getState();
+
+    // Convenience references to commonly used properties.
+    const SData& request = command->request;
+    SData& response = command->response;
+    STable& content = command->jsonContent;
+
+    // We catch any exception and handle in `_handleCommandException`.
+    RESULT returnValue = RESULT::COMPLETE;
+    try {
+        SDEBUG("prePeeking at '" << request.methodLine << "' with priority: " << command->priority);
+        uint64_t timeout = _getRemainingTime(command, false);
+        command->prePeekCount++;
+
+        _db.startTiming(timeout);
+
+        try {
+            if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
+                STHROW("501 Failed to begin shared prePeek transaction");
+            }
+
+            // Make sure no writes happen while in prePeek command
+            _db.setQueryOnly(true);
+
+            // prePeek.
+            command->reset(BedrockCommand::STAGE::PREPEEK);
+            bool completed = command->prePeek(_db);
+            SDEBUG("Plugin '" << command->getName() << "' prePeeked command '" << request.methodLine << "'");
+
+            if (command->httpsRequests.size()) {
+                STHROW("405 https requests cannot be made in prePeek");
+            }
+
+            if (!completed) {
+                SDEBUG("Command '" << request.methodLine << "' not finished in prePeek, re-queuing.");
+                _db.rollback();
+                _db.resetTiming();
```

Comment:
I don't think we want this line here?

Reply:
I THINK we do, but I'm not 100% sure. If you look at `peekCommand`, when the command is not completed in `peek` and we return `SHOULD_PROCESS` we call the `_db.resetTiming()` just before hand. That is similar to what we are doing here, no?

https://github.com/Expensify/Bedrock/blob/main/BedrockCore.cpp#L101